packages/core/test/lib/integrations/inboundfilters.test.ts

```diff
@@ -1,522 +1,314 @@
-          inboundFilters._isIgnoredError(
-            exceptionEvent,
-            inboundFilters._mergeOptions({
-              ignoreErrors: [/unidentified \?/],
-            }),
-          ),
-        ).toBe(true);
+        const eventProcessor = createInboundFiltersEventProcessor({
+          ignoreErrors: [/unidentified \?/],
+        });
+        expect(eventProcessor(EXCEPTION_EVENT)).toBe(null);
       });
 
       it('can match on exception type', () => {
-        expect(
-          inboundFilters._isIgnoredError(
-            exceptionEvent,
-            inboundFilters._mergeOptions({
-              ignoreErrors: [/^SyntaxError/],
-            }),
-          ),
-        ).toBe(true);
+        const eventProcessor = createInboundFiltersEventProcessor({
+          ignoreErrors: [/^SyntaxError/],
+        });
+        expect(eventProcessor(EXCEPTION_EVENT)).toBe(null);
       });
     });
   });
 
   describe('denyUrls/allowUrls', () => {
-    const messageEvent = {
-      message: 'wat',
-      stacktrace: {
-        // Frames are always in the reverse order, as this is how Sentry expect them to come.
-        // Frame that crashed is the last one, the one from awesome-analytics
-        frames: [
-          { filename: 'https://our-side.com/js/bundle.js' },
-          { filename: 'https://our-side.com/js/bundle.js' },
-          { filename: 'https://awesome-analytics.io/some/file.js' },
-        ],
-      },
-    };
-    const exceptionEvent = {
-      exception: {
-        values: [
-          {
-            stacktrace: {
-              // Frames are always in the reverse order, as this is how Sentry expect them to come.
-              // Frame that crashed is the last one, the one from awesome-analytics
-              frames: [
-                { filename: 'https://our-side.com/js/bundle.js' },
-                { filename: 'https://our-side.com/js/bundle.js' },
-                { filename: 'https://awesome-analytics.io/some/file.js' },
-              ],
-            },
-          },
-        ],
-      },
-    };
-
     it('should filter captured message based on its stack trace using string filter', () => {
-      expect(
-        inboundFilters._isDeniedUrl(
-          messageEvent,
-          inboundFilters._mergeOptions({
-            allowUrls: ['https://awesome-analytics.io'],
-            denyUrls: ['https://awesome-analytics.io'],
-          }),
-        ),
-      ).toBe(true);
-      expect(
-        inboundFilters._isAllowedUrl(
-          messageEvent,
-          inboundFilters._mergeOptions({
-            allowUrls: ['https://awesome-analytics.io'],
-            denyUrls: ['https://awesome-analytics.io'],
-          }),
-        ),
-      ).toBe(true);
+      const eventProcessorBoth = createInboundFiltersEventProcessor({
+        allowUrls: ['https://awesome-analytics.io'],
+        denyUrls: ['https://awesome-analytics.io'],
+      });
+      expect(eventProcessorBoth(MESSAGE_EVENT_WITH_STACKTRACE)).toBe(null);
```

Comment:
This feels like it belongs in a separate test, proving that `denyUrls` is stronger than `allowUrls`. (Though, to be honest, I don't think this is a case worth testing, since no one should be putting the same exact value into both. More to the point would be to test when one pattern's matches are a subset of the other pattern's matches, in both directions.)

Reply:
Separated into different tests